tests/testthat/test-public_api.R

```diff
@@ -17,3 +17,37 @@ test_that("styler can style file", {
 })
 
 # style_active_file() must be tested manually.
+test_that("...", {
+  before_styling <- utf8::read_lines_enc(paste0(base, "/xyzaddin/addin_region-in.R"))
+  reference <- utf8::read_lines_enc(paste0(base, "/xyzaddin/addin_region-out.R"))
+  result <- expect_error(
+    testthat::with_mock(
```

Comment:
`mockr::with_mock()` should really work here, because `testthat::with_mock()` might not at some point. Can you please simplify the code (e.g., extract the hard-coded `"document_context"` structure to a variable) and try again? For troubleshooting the error message is helpful.

Reply:
`mockr::with_mock()` works using `devtools::test()` but not interactively while `testthat::with_mock()` works in both sistuations, which is why I have not used `mockr` yet. Can you have a look at why that is the case?

What do you mean with extracting hard-coded `"document_context"` structure to a variable?  Like this?

```r
doc_cont <- structure(...)
testthat("...", {
  result <- mockr::with_mock(
    get_rstudio_context = function() doc_cont, 
    ...
```
Unfortunately, there is no way to create the structure with a function call directly.

Reply:
Yes, something like this code.

You could try `mockr::with_mock(..., envir = asNamespace("styler"))` .

Reply:
Ok, so assign the variable elsewhere in the name space? Or should leave the declaration where it is?